packages/nextjs/src/server/content-security-policy.ts

```diff
@@ -0,0 +1,357 @@
+/**
+ * Type representing valid CSP directives
+ */
+export type CSPDirective =
+  // Default resource directives
+  | 'connect-src'
+  | 'default-src'
+  | 'font-src'
+  | 'img-src'
+  | 'media-src'
+  | 'object-src'
+  | 'script-src'
+  | 'style-src'
+  // Framing and navigation directives
+  | 'base-uri'
+  | 'child-src'
+  | 'form-action'
+  | 'frame-ancestors'
+  | 'frame-src'
+  | 'manifest-src'
+  | 'navigate-to'
+  | 'prefetch-src'
+  | 'worker-src'
+  // Sandbox and plugin directives
+  | 'plugin-types'
+  | 'require-sri-for'
+  | 'sandbox'
+  // Trusted types and upgrade directives
+  | 'block-all-mixed-content'
+  | 'require-trusted-types-for'
+  | 'trusted-types'
+  | 'upgrade-insecure-requests'
+  // Reporting directives
+  | 'report-to'
+  | 'report-uri'
+  // CSP Level 3 additional directives
+  | 'script-src-attr'
+  | 'script-src-elem'
+  | 'style-src-attr'
+  | 'style-src-elem';
+
+/**
+ * Type representing the CSP mode
+ */
+export type CSPMode = 'standard' | 'strict-dynamic';
+
+/**
+ * Type representing CSP values as a record of directives to string arrays
+ */
+type CSPValues = Partial<Record<CSPDirective, string[]>>;
+
+/**
+ * Type representing CSP directives as a record of directives to Sets of strings
+ */
+type CSPDirectiveSet = Record<CSPDirective, Set<string>>;
+
+/**
+ * Interface representing the result of creating a CSP header
+ */
+export interface CSPHeaderResult {
+  /** The formatted CSP header string */
+  header: string;
+  /** The generated nonce, if applicable */
+  nonce?: string;
+}
+
+/**
+ * Class responsible for managing CSP directives and their values
+ */
+class CSPDirectiveManager {
+  /** Set of special keywords that require quoting in CSP directives */
+  private static readonly KEYWORDS = new Set([
+    'none',
+    'self',
+    'strict-dynamic',
+    'unsafe-eval',
+    'unsafe-hashes',
+    'unsafe-inline',
+  ]);
+
+  /** Default CSP directives and their values */
+  static readonly DEFAULT_DIRECTIVES: CSPValues = {
+    'connect-src': ['self'],
+    'default-src': ['self'],
+    'form-action': ['self'],
+    'frame-src': ['self', 'https://challenges.cloudflare.com'],
+    'img-src': ['self', 'https://img.clerk.com'],
+    'script-src': [
+      'self',
+      ...(process.env.NODE_ENV !== 'production' ? ['unsafe-eval'] : []),
+      'unsafe-inline',
+      'https:',
+      'http:',
+    ],
+    'style-src': ['self', 'unsafe-inline'],
+    'worker-src': ['self', 'blob:'],
+  };
```

Comment:
Just noting for follow up: we'll need to add stripe stuff for commerce here soon

Reply:
Should I just add that in this PR?

Reply:
I suppose it doesn't hurt to add it now 👍

Reply:
Alrighty!